distributed/client.py

```diff
@@ -1470,7 +1470,8 @@ def submit(
         pure : bool (defaults to True)
             Whether or not the function is pure.  Set ``pure=False`` for
             impure functions like ``np.random.random``.
-            See :ref:`pure functions` for more details.
+            See `pure functions <https://distributed.dask.org/en/latest/client.html#pure-functions>`_
+            for more details.
```

Comment:
I'm curious if these `:ref:`s were causing warning (I built the docs locally on the current `main` branch, but didn't see any warnings related to this). While what you have here is valid, the current `:ref:` approach is nice because it works with both the `stable` and `latest` version of the documentation which point to the latest released version and development version of the docs, respectively. With the changes in this PR, these links will always redirect to the `latest` documentation version, even when originally viewing the `stable` version.

Reply:
Ah, I missed that you mentioned why these changes were added in your original comment (to avoid, for example, `unknown document` warnings when the Dask documentation is built)

Reply:
Good point about `stable` vs. `latest`. The `unknown document` warnings actually come up in the Dask docs build, not the `distributed` docs build. This is because the Dask API autodocs are actually referring to some of the `distributed` methods and mixins.

I tried to see if there was a way to avoid or ignore those errors when building from the Dask docs, but I couldn't find anything in Sphinx that would specifically let us ignore these and not hide others in the future. We could also vendor or submodule `distributed` in the Dask repo, but that seems like overkill for ~10 docstring cross-references. We could also move the cross-references out of the docstrings and put them in `distributed` docs, but I wanted to aim for a solution that would give us links in both Dask docs and `distributed` docs.

Let me know if you are okay with the absolute link approach, or if you want to look into another approach. In this PR, I also prioritized getting clean dask and distributed docs builds, whereas we can look for other solutions for these cross references in a future PR.

Reply:
Oops, after re-reading your comment, I just realized that I used `latest` instead of `stable`. Changed the links to `stable` in 746d3a7, which is a more sensible default.